covid_graphs/covid_graphs/scripts/server.py

```diff
@@ -36,31 +42,53 @@ def run_server(data_dir: Path, simulated_polynomial: Path, simulated_exponential
     """Creates and runs a flask server."""
     server = Flask(__name__, template_folder=CURRENT_DIR)
 
-    covid19_normal_app = CountryGraph.create_dashboard(data_dir, server, GraphType.Normal)
-    covid19_semilog_app = CountryGraph.create_dashboard(data_dir, server, GraphType.SemiLog)
-    covid19_loglog_app = CountryGraph.create_dashboard(data_dir, server, GraphType.LogLog)
-    covid19_heatmap_app = HeatMap(str(simulated_polynomial)).create_app(server)
-    covid19_heatmap_exponential_app = HeatMap(str(simulated_exponential)).create_app(server)
+    _create_prediction_apps(server=server, data_dir=data_dir)
+    _create_simulation_apps(server, simulated_polynomial, simulated_exponential)
 
     @server.route("/")
     def home():
         return render_template("index.html")
 
     @server.route("/covid19/")
     def covid19_redirect():
-        return redirect(url_for("covid19_normal"))
+        return redirect(url_for("covid19_predictions", event="apr12", graph_type="normal"))
+
+    server.run(host="0.0.0.0", port=8081, debug=True, extra_files=data_dir.glob("*.data"))
+
 
-    @server.route("/covid19/normal")
-    def covid19_normal():
-        return covid19_normal_app.index()
+def _create_prediction_apps(data_dir: Path, server: Flask):
+    # Note: Event route strings might become part the PredictionEvent class.
+    event_by_route = {
+        "mar29": predictions.BK_20200329,
+        "apr12": predictions.BK_20200412,
+    }
+    graph_type_by_route = {
+        "normal": GraphType.Normal,
+        "semilog": GraphType.SemiLog,
+        "loglog": GraphType.LogLog,
+    }
+    route_pairs = itertools.product(event_by_route.keys(), graph_type_by_route.keys())
 
-    @server.route("/covid19/semilog")
-    def covid19_semilog():
-        return covid19_semilog_app.index()
+    prediction_apps = {
+        (event_route, graph_type_route): country_dashboard.create_dashboard(
```

Comment:
What sourcery is this? A lambda without the `lambda` keyword?

Reply:
LOL :D No, it is a dictionary comprehension:
```py
dct = {key: val for x in whatever}
```

Reply:
`prediction_apps` is of type `Dict[Tuple[str, str], Dash]`. I create the dash apps before registering routes with flask, this is where I store them. It is not super elegant. I think it might be not necessary though, did not investigate.

Reply:
Oh, dicitonary comprehension, I forgot that exists!